libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccountManager.m

```diff
@@ -163,9 +192,24 @@ - (BOOL)idpEnabled {
 }
 
 - (void)setIdpEnabled:(BOOL)idpEnabled {
+    if (idpEnabled) {
+        [SFSDKAppFeatureMarkers registerAppFeature:kSFSPAppFeatureIDPLogin];
+    }
     self.authPreferences.idpEnabled = idpEnabled;
 }
 
+- (SFOAuthAdvancedAuthConfiguration)advancedAuthConfiguration {
+   return self.authPreferences.advancedAuthConfiguration;
+}
+
+- (void)setAdvancedAuthConfiguration:(SFOAuthAdvancedAuthConfiguration)advancedAuthConfiguration {
+    if (advancedAuthConfiguration == SFOAuthAdvancedAuthConfigurationRequire)
```

Comment:
You need an `else` block too that calls `[SFSDKAppFeatureMarkers removeAppFeature:kSFAppFeatureSafariBrowserForLogin];`, since the flag can be toggled the other way and we'd have to remove the existing app marker then. This can happen when a new value is pushed down through MDM that enables/disables this feature.

Reply:
You also need some more logic here for the `SFOAuthAdvancedAuthConfigurationAllow` use case that adds this marker if the swizzle occurs. I'd suggest removing this marker from here and setting it when we instantiate `SFSafariViewController` instead.